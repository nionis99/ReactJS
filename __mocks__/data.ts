export const genres = ['Crime', 'Documentary', 'Horror', 'Comedy'];
export const MovieMockData = {
  id: 0,
  title: 'The Avengers 2',
  vote_average: 7.5,
  release_date: '2012-04-25',
  poster_path: 'https://image.tmdb.org/t/p/w500/cezWGskPY5x7GaglTTRN4Fugfb8.jpg',
  overview: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
  genres: ['Science Fiction', 'Action', 'Adventure'],
  runtime: 143,
};
