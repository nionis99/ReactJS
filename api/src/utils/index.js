const generateId = () => Date.now();

const parseQuery = (ctx, next) => {
  ctx.query.offset = (ctx.query.offset && parseInt(ctx.query.offset, 10)) || 0;
  ctx.query.limit = (ctx.query.limit && parseInt(ctx.query.limit, 10)) || 12;

  next();
};

module.exports = {
  generateId,
  parseQuery,
};
