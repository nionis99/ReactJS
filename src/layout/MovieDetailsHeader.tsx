import React, { HTMLAttributes } from 'react';
import moment from 'moment';
import useStateSelector from 'hooks/useStateSelector';
import Index from 'components/Loader';
import SearchButton from 'assets/icons/searchButton.svg';
import NotFoundImage from 'assets/images/not_found.png';
import { Movie } from 'reducers/movieReducer/types';

interface MovieDetailsHeaderProps extends HTMLAttributes<HTMLDivElement> {
  onSearchClick: () => void;
  movie: Movie;
}

const MovieDetailsHeader = ({ onSearchClick, movie, className = '', ...rest }: MovieDetailsHeaderProps) => {
  const { getMovieLoading } = useStateSelector((state) => state.movies);
  const { poster_path, title, genres, release_date, overview, vote_average, runtime } = movie;
  const yearsOfTheMovie = moment(release_date).format('YYYY');
  const durationInHours = `${(runtime / 60).toFixed(0)}h ${(runtime % 60).toString()}min`;

  return (
    <header className={`flex-col bg-content py-2 md:py-5 px-4 md:px-16 text-white  ${className}`} {...rest}>
      {getMovieLoading ? (
        <Index className="w-32 h-32" />
      ) : (
        <>
          <div className="flex w-full justify-between items-center">
            <span className="font-light text-primary">netflixroulette</span>
            <span className="ml-auto">
              <SearchButton className="cursor-pointer" onClick={onSearchClick} />
            </span>
          </div>
          <div className="flex flex-col md:flex-row md:justify-center pt-6">
            <div className="flex w-full md:w-1/3 justify-center items-center h-auto">
              <img
                alt={title}
                src={poster_path}
                width={400}
                height={200}
                onError={(event) => (event.currentTarget.src = NotFoundImage)}
              />
            </div>
            <div className="w-full md:w-2/3 px-8">
              <p className="flex font-light justify-center md:justify-start pt-3 md:pt-0">
                <span className="flex justify-center items-center text-4xl">{title}</span>
                <span
                  className="flex ml-8 items-center justify-center p-3 h-14 w-14 rounded-full border-2 border-white
              text-2xl"
                >
                  {vote_average}
                </span>
              </p>
              <p className="flex text-white text-opacity-50 justify-center md:justify-start pt-3 md:pt-0">
                {genres.join(', ')}
              </p>
              <p className="flex py-8 text-primary font-light justify-center md:justify-start text-2xl">
                <span className="mr-8">{yearsOfTheMovie}</span>
                <span>{durationInHours}</span>
              </p>
              <p className="text-white text-opacity-50">{overview}</p>
            </div>
          </div>
        </>
      )}
    </header>
  );
};

export default MovieDetailsHeader;
