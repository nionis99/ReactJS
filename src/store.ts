import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { MovieActionsTypes } from 'actions/movieActions/types';
import rootReducer from 'reducers';
import thunk from 'redux-thunk';

const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)));

export type RootState = ReturnType<typeof rootReducer>;
export type RootActions = MovieActionsTypes;
export default store;
