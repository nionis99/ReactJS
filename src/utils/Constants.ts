export const ROUTES = {
  search: '/search',
};

export enum testingConstants {
  addMovieButton,
  searchHeaderInput,
  searchSubmitButton,
  moviesList,
  movieItemOptions,
  editMovieOption,
  removeMovieOption,
  confirmRemoveMovie,
  genresOptionsInput,
  movieOverviewTextarea,
  submitMovieForm,
  movieCard,
  searchButton,
}

const constants = {
  headerTitle: 'FIND YOUR MOVIE',
};

export default constants;
