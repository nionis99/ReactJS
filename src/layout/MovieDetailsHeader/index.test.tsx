import '@testing-library/jest-dom';
import React from 'react';
import moment from 'moment';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { fireEvent, render } from '@testing-library/react';
import { MovieMockData } from '../../../__mocks__/data';
import { testingConstants } from 'utils/Constants';
import store from 'store';
import MovieDetailsHeader from '.';

describe('Movie details header component', () => {
  const onClick = jest.fn();
  const wrapper = (
    <Provider store={store}>
      <BrowserRouter>
        <MovieDetailsHeader movie={MovieMockData} onSearchClick={onClick} />
      </BrowserRouter>
    </Provider>
  );

  it('should render movie details header component', () => {
    const elem = renderer.create(wrapper).toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('should be clicked and invoked onClick function passed in props', () => {
    const { getByText, getByTestId } = render(wrapper);
    expect.assertions(5);
    expect(getByText(MovieMockData.title)).toBeInTheDocument();
    expect(getByText(moment(MovieMockData.release_date).format('YYYY'))).toBeInTheDocument();
    expect(getByText(MovieMockData.overview)).toBeInTheDocument();
    expect(getByText(MovieMockData.vote_average)).toBeInTheDocument();
    fireEvent.click(getByTestId(testingConstants.searchButton));
    expect(onClick).toBeCalledTimes(1);
  });
});
