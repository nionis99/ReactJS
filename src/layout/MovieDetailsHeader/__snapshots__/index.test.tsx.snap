// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie details header component should render movie details header component 1`] = `
<header
  className="flex-col bg-content py-2 md:py-5 px-4 md:px-16 text-white  "
>
  <div
    className="flex w-full justify-between items-center"
  >
    <span
      className="font-light text-primary"
    >
      netflixroulette
    </span>
    <span
      className="ml-auto"
    >
      <test-file-stub
        className="cursor-pointer"
        data-testid={12}
        onClick={[MockFunction]}
      />
    </span>
  </div>
  <div
    className="flex flex-col md:flex-row md:justify-center pt-6"
  >
    <div
      className="flex w-full md:w-1/3 justify-center items-center h-auto"
    >
      <img
        alt="The Avengers 2"
        height={200}
        onError={[Function]}
        src="https://image.tmdb.org/t/p/w500/cezWGskPY5x7GaglTTRN4Fugfb8.jpg"
        width={400}
      />
    </div>
    <div
      className="w-full md:w-2/3 px-8"
    >
      <p
        className="flex font-light justify-center md:justify-start pt-3 md:pt-0"
      >
        <span
          className="flex justify-center items-center text-4xl"
        >
          The Avengers 2
        </span>
        <span
          className="flex ml-8 items-center justify-center p-3 h-14 w-14 rounded-full border-2 border-white
              text-2xl"
        >
          7.5
        </span>
      </p>
      <p
        className="flex text-white text-opacity-50 justify-center md:justify-start pt-3 md:pt-0"
      >
        Science Fiction, Action, Adventure
      </p>
      <p
        className="flex py-8 text-primary font-light justify-center md:justify-start text-2xl"
      >
        <span
          className="mr-8"
        >
          2012
        </span>
        <span>
          2h 23min
        </span>
      </p>
      <p
        className="text-white text-opacity-50"
      >
        Lorem Ipsum is simply dummy text of the printing and typesetting industry.
      </p>
    </div>
  </div>
</header>
`;
